### PR TITLE
portalocker 3.0.0

### PR DESCRIPTION
**Pull request by psteyer**
portalocker 3.0.0

**Destination channel:** defaults

### Links

- [PKG-6510](https://anaconda.atlassian.net/browse/PKG-6510) 
- [Upstream repository](https://github.com/WoLpH/portalocker/tree/v3.0.0  )
- Relevant dependency PRs:
  - https://github.com/AnacondaRecipes/torchdata-feedstock/pull/4

### Explanation of changes:

- building 3.0.0 - for `torchdata`
- added back in tests

[PKG-6510]: https://anaconda.atlassian.net/browse/PKG-6510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,11 +8,12 @@ package:
 source:
   fn: {{ name }}-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 4e913d807aa6598c320e8a50c50e2ee0602bc45240b485e3f8bc06f13060084c
+  sha256: 21f535de2e7a82c94c130c054adb5c7421d480d5619d61073996e2f89bcb879b
 
 build:
   number: 1
```
Comment (cbouss) [inline recipe/meta.yaml:14]:
```suggestion
  number: 0
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -8,11 +8,12 @@ package:
...
   script: {{ PYTHON }} -m pip install . --no-deps -vv
+  skip: true  # [py<38 or py>312]
```
Comment (cbouss) [inline recipe/meta.yaml:16]:
```suggestion
  skip: true  # [py<39]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -8,11 +8,12 @@ package:
 source:
   fn: {{ name }}-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 4e913d807aa6598c320e8a50c50e2ee0602bc45240b485e3f8bc06f13060084c
+  sha256: 21f535de2e7a82c94c130c054adb5c7421d480d5619d61073996e2f89bcb879b
 
 build:
   number: 1
   script: {{ PYTHON }} -m pip install . --no-deps -vv
```
Comment (cbouss) [inline recipe/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -8,11 +8,12 @@ package:
...
+  skip: true  # [py<38 or py>312]
 
 requirements:
   host:
```
Comment (cbouss) [inline recipe/meta.yaml:19]: requires seutptools and setuptoools-scm in host
pywin32 >=226 in run

File: recipe/meta.yaml [hunk 5]
```diff
@@ -33,7 +34,7 @@ test:
     - pytest-cache >=1.0
```
Comment (cbouss) [inline recipe/meta.yaml:34]: Since not running the tests, no need to add pytest. 
pip check would be welcome.

File: recipe/meta.yaml (hunk 1 above)
Comment (psteyer) [reply recipe/meta.yaml]: done!

File: recipe/meta.yaml (hunk 2 above)
Comment (psteyer) [reply recipe/meta.yaml]: done!

File: recipe/meta.yaml (hunk 3 above)
Comment (psteyer) [reply recipe/meta.yaml]: done

File: recipe/meta.yaml (hunk 5 above)
Comment (psteyer) [reply recipe/meta.yaml]: done!

File: recipe/meta.yaml [hunk 6]
```diff
@@ -8,35 +8,32 @@ package:
...
-    - portalocker_tests
   requires:
-    - pytest >=3.4.0
-    - pytest-cache >=1.0
-    - pytest-cov >=2.5.1
-    - pytest-flakes >=2.0.0
-    - pytest-pep8 >=1.0.6
-  # TODO: how to skip just test_combined ?
-  # commands:
-  #  - pytest -k "not test_combined" portalocker_tests
```
Comment (Jrice1317) [inline recipe/meta.yaml:39]: `pytest --ignore=test_combined.py -v portalocker_tests` should work. That or something like `pytest --ignore=portalocker_tests/test_combined.py -v portalocker_tests`

File: recipe/meta.yaml [hunk 7]
```diff
@@ -8,35 +8,32 @@ package:
...
+    - pip
+  commands:
+    - pip check
 
 about:
   home: https://github.com/WoLpH/portalocker
```
Comment (Jrice1317) [inline recipe/meta.yaml:39]: Can we please add a description and change the `doc_url` to the `https`?

File: recipe/meta.yaml (hunk 4 above)
Comment (psteyer) [reply recipe/meta.yaml:19]: went ahead and added wheel also

File: recipe/meta.yaml (hunk 7 above)
Comment (psteyer) [reply recipe/meta.yaml:44]: Fixed!

File: recipe/meta.yaml (hunk 6 above)
Comment (psteyer) [reply recipe/meta.yaml:39]: added tests back in!

File: recipe/meta.yaml [hunk 8]
```diff
@@ -8,43 +8,45 @@ package:
...
   run:
     - python
-    - pywin32  # [win]
+    - pywin32 >=226  # [win]
 
 test:
-  imports:
-    - portalocker
   source_files:
     - portalocker_tests
```
Comment (Jrice1317) [inline recipe/meta.yaml:32]:
```suggestion
    - portalocker_tests
    - portalocker/redis.py
```

File: recipe/meta.yaml [hunk 9]
```diff
@@ -8,43 +8,45 @@ package:
...
+  commands:
+    - pip check
+    - pytest -v portalocker_tests --ignore=portalocker_tests/test_combined.py
 
 about:
   home: https://github.com/WoLpH/portalocker
   license: PSF-2.0
   license_family: PSF
   license_file: LICENSE
   summary: Portalocker is a library to provide an easy API to file locking.
```
Comment (Jrice1317) [inline recipe/meta.yaml:48]:
```suggestion
  summary: Portalocker is a library to provide an easy API to file locking.
  description: |
    - An easy library for Python file locking. It works on Windows, Linux, BSD and Unix systems and can even perform distributed locking. Naturally it also supports the with statement.
```

File: recipe/meta.yaml [hunk 10]
```diff
@@ -8,43 +8,45 @@ package:
...
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+  skip: true  # [py<39]
 
 requirements:
   host:
     - python
     - pip
     - pytest-runner
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:22]: I think pytest-runner is not needed, see https://github.com/wolph/portalocker/blob/v3.0.0/pyproject.toml#L3 and https://github.com/wolph/portalocker/commit/05331bfa0aeb06b29eb7a28510466de92c438d5e

File: recipe/meta.yaml (hunk 9 above)
Comment (psteyer) [reply recipe/meta.yaml:48]: Forgot to push the description. Thank you!

File: recipe/meta.yaml (hunk 10 above)
Comment (psteyer) [reply recipe/meta.yaml:22]: removed

File: recipe/meta.yaml (hunk 8 above)
Comment (psteyer) [reply recipe/meta.yaml:32]: Never mind! I think I understand. It seems `portalocker/redis.py` isn't satisfying the testing requirement either.

**Comment by skupr-anaconda** [comment]
Tests failed on linux-64, linux-aarch64, and osx-64:
`E   ModuleNotFoundError: No module named 'redis'`

**Review by skupr-anaconda — APPROVED** [review]

**Review by Jrice1317 — APPROVED** [review]
Nice work!

**Pull request merged**